src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -1069,5 +1074,27 @@ public UUID getSnapshotBillingProfileId() {
     public CloudPlatform getCloudPlatform() {
       return this.cloudPlatform;
     }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      sb.append("class SnapshotCacheResult {\n");
+      sb.append("  snapshot: {\n");
+      sb.append("    id: ").append(this.id).append("\n");
+      sb.append("    name: ").append(this.name).append("\n");
+      sb.append("    isSelfHosted: ").append(this.isSelfHosted).append("\n");
+      sb.append("    globalFileIds: ").append(this.globalFileIds).append("\n");
+      sb.append("    billingProfileId: ").append(this.snapshotBillingProfileId).append("\n");
+      sb.append("    cloudPlatform: ").append(this.cloudPlatform).append("\n");
+      sb.append("    googleProjectId: ").append(this.googleProjectId).append("\n");
+      sb.append("  }\n");
+      sb.append("  rootDataset: {\n");
+      sb.append("    id: ").append(this.datasetId).append("\n");
+      sb.append("    name: ").append(this.datasetName).append("\n");
+      sb.append("    datasetProjectId: ").append(this.datasetProjectId).append("\n");
+      sb.append("  }\n");
+      sb.append("}");
+      return sb.toString();
+    }
```

Comment:
It's longer but I think a text block makes it clearer what the output will look like:
```
    @Override
    public String toString() {
      return """
          class SnapshotCacheResult {
            snapshot: {
              id: %s
              name: %s
              isSelfHosted: %s
              globalFileIds: %s
              billingProfileId: %s
              cloudPlatform: %s
              googleProjectId: %s
            }
            rootDataset: {
              id: %s
              name: %s
              datasetProjectId: %s
            }
          }
          """
          .formatted(
              id,
              name,
              isSelfHosted,
              globalFileIds,
              snapshotBillingProfileId,
              cloudPlatform,
              googleProjectId,
              datasetId,
              datasetName,
              datasetProjectId);
    }
```

Reply:
I agree with you.  I'll resolve this convo as it's no longer needed with a refactor to a record.